### PR TITLE
Revise model by changing the economic model and the pressure drop implementation

### PR DESCRIPTION
**Pull request by iakkor**
## Changes proposed in this PR:
- update to the pressure drop model in packed columns
- new economic model

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my
contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file
   at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: akkor_et_al_2024/Advanced_Flash_Stripper.py [hunk 1]
```diff
@@ -241,9 +241,16 @@ def initialize(blk, solver, outlvl):
...
         # packing parameters for the stripper (random packing)
         blk.stripper.del_component(blk.stripper.a)
-        blk.stripper.a = Param(initialize=250, doc="Specific area (m2/m3)")
+        blk.stripper.a = Param(
+            initialize=250, mutable=True, doc="Specific area (m2/m3)"
+        )
         blk.stripper.del_component(blk.stripper.void)
-        blk.stripper.void = Param(initialize=0.97, doc="Void fraction")
+        blk.stripper.void = Param(initialize=0.97, mutable=True, doc="Void fraction")
+        blk.stripper.del_component(blk.stripper.P_drop_z)
```
Comment (andrewlee94) [inline akkor_et_al_2024/Advanced_Flash_Stripper.py]: Minor question - why are you deleting this (and on line 247 too)?

File: akkor_et_al_2024/PZ_AFS_flowsheet.py [hunk 2]
```diff
@@ -357,8 +358,8 @@ def connect_to_abs2(m):
                 )
 
     # report results
-    for v in m.component_data_objects(Var):
-        print(v.name + ": " + str(v.value))
+    # for v in m.component_data_objects(Var):
```
Comment (andrewlee94) [inline akkor_et_al_2024/PZ_AFS_flowsheet.py]: Probably best to remove the commented code.

**Review by andrewlee94 — APPROVED** [review]

File: akkor_et_al_2024/PZ_AFS_flowsheet.py (hunk 2 above)
Comment (iakkor) [reply akkor_et_al_2024/PZ_AFS_flowsheet.py]: Thanks, I removed it and added code to report the relevant results only.

File: akkor_et_al_2024/Advanced_Flash_Stripper.py (hunk 1 above)
Comment (iakkor) [reply akkor_et_al_2024/Advanced_Flash_Stripper.py]: These values should be different for the stripper column. In my experience, it is safer to redefine the parameter than to change its value during initialization. In the past I had some issues where the expressions referencing the parameters were not updated correctly.

Comment (andrewlee94) [reply akkor_et_al_2024/Advanced_Flash_Stripper.py]: Deleting components will actually make things more of a mess doing it this way, as any references will remain but be hidden. You should be able to change the values of these without any issues, and if you do see problems that generally implies you are doing something wrong elsewhere.

Seeing as this is part of the an initialization routine, why are you changing the parameter values here? Setting parameter values is something that should be done whilst defining the the model (i.e. before you start initialization), not when initializing it.

**Comment by andrewlee94** [comment]
@iakkor Do you have the ability to merge this PR, or do you need me to do so? If I need to do it, let me know when you think it is ready.

**Comment by iakkor** [comment]
> @iakkor Do you have the ability to merge this PR, or do you need me to do so? If I need to do it, let me know when you think it is ready.

I don't. I'll let you know once it is finalized, thanks!

**Comment by andrewlee94** [comment]
@iakkor Before I forget - I will be leaving NETL for a new job on the 6th. No need to rush, but you might need to contact someone else to merge the PR after that.

**Comment by iakkor** [comment]
> @iakkor Before I forget - I will be leaving NETL for a new job on the 6th. No need to rush, but you might need to contact someone else to merge the PR after that.

It is ready to be merged. Thanks for everything Andrew!

**Pull request merged**